### PR TITLE
Adding suport to Port Map (PMAP) module

### PR DESCRIPTION
**Pull request by JoseClaudioSJr**

### CODE REVIEW

**Comment by reitermarkus** [comment]
Did you see https://github.com/msp432-rust/msp432p401r-hal/pull/5?

I haven't had time to rebase it yet and reopen a PR.

**Comment by JoseClaudioSJr** [comment]
> Did you see #5?
> 
> I haven't had time to rebase it yet and reopen a PR.

@reitermarkus Yes, I took this code as a basis. Some remarks:
- Only the ports P2, P3  and P7 can be remapped. According to the datasheet: "Up to 24 I/Os with port mapping capability". "6.9.2.1 Port Mapping Definitions - The port mapping controller on MSP432P401x MCUs allows reconfigurable mapping of digital functions on ports P2, P3, and P7";
- "Interrupts should be disabled during the configuration process";
- "If it is necessary to reconfigure the mapping during runtime, the PMAPRECFG bit must be set during the first write access timeslot. If PMAPRECFG is cleared during later configuration sessions, no more configuration sessions are possible". Therefore, the PMAPRECFG bit needs to be set only once, having an effect for the entire port map.
- Finally, we can discuss whether it is better to configure the remap in a GPIO routine or externally in a PMAP routine. In C code, they are separate routines. But, it seems indifferent to the user.
(https://github.com/energia/Energia/blob/master/hardware/msp432/cores/msp432/driverlib/pmap.c)

**Comment by reitermarkus** [comment]
> Only the ports P2, P3 and P7 can be remapped.

Yes, I hadn't looked specifically at which ports are supported, merely made it so it compiles.

> In C code, they are separate routines.

Is there any use case for mapping a port without also using a GPIO pin? If not, I think having the method on the pin itself is a bit rustier. After all, just because something is some way in C doesn't mean it's the best abstraction for the job.

**Comment by JoseClaudioSJr** [comment]
@caiotavares

File: examples/spi.rs [hunk 1]
```diff
@@ -66,9 +65,9 @@ fn main() -> ! {
         .with_bit_rate_prescaler(0x02);
 
     // Setup eUSCI_A1 SPI PINs into proper alternate mode
-    gpio.p2_0.into_alternate_primary();
-    gpio.p2_1.into_alternate_primary();
-    gpio.p2_2.into_alternate_primary();
+    gpio.p2_4.into_alternate_primary().remap(Mapping::UCA1STE, true);
+    gpio.p2_6.into_alternate_primary().remap(Mapping::UCA1CLK,true);
+    gpio.p2_7.into_alternate_primary().remap(Mapping::UCA1SOMI, true);
```
Comment (caiotavares) [inline examples/spi.rs:70]: Pretty cool!

File: src/pmap.rs [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    fn key_lock(&mut self, lock: bool) -> &Self{
+        if lock == false {
+            self.map.as_ref().unwrap().pmapkeyid.write(|w| unsafe {
+                w.pmapkey().bits(UNLOCK_KEY)
+            });
+        } else {
+            self.map.as_ref().unwrap().pmapkeyid.write(|w| unsafe {
+                w.pmapkey().bits(0)
+            });
+        }
```
Comment (caiotavares) [inline src/pmap.rs]:
```suggestion
        if lock {
            self.map.as_ref().unwrap().pmapkeyid.write(|w| unsafe {
                w.pmapkey().bits(0)
            });
        } else {
            self.map.as_ref().unwrap().pmapkeyid.write(|w| unsafe {
                w.pmapkey().bits(UNLOCK_KEY)
            });
        }
```

**Review by caiotavares — DISMISSED** [review]

**Review by caiotavares — APPROVED** [review]

**Pull request merged**